lib/jazzy/source_declaration/type.rb

```diff
@@ -61,6 +61,14 @@ def extension?
         kind =~ /^source\.lang\.swift\.decl\.extension.*/
       end
 
+      def extensible?
+        kind =~ /^source\.lang\.swift\.decl\.(class|struct|protocol|enum)$/
+      end
+
+      def protocol?
+        kind =~ /^source\.lang\.swift\.decl\.protocol$/
```

Comment:
it seems nicer/cleaner to just do a string comparison if we're matching for an exact string anyway?